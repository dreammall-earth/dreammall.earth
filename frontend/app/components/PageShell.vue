<template>
  <v-app>
    <slot />
  </v-app>
</template>
