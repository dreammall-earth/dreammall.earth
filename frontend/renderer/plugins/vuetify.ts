// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
