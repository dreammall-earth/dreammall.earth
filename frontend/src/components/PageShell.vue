<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
