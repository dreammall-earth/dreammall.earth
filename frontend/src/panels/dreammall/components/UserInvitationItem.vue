<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
