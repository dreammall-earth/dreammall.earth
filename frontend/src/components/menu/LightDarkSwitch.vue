<template>
  <Switch
    class="light-dark-switch"
    @change="setTheme"
    :defaultState="theme.global.current.value.dark ? 'right' : 'left'"
    :label="`${$t('menu.theme.light')} / ${$t('menu.theme.dark')}`"
  />
</template>

<script lang="ts" setup>
import { useTheme } from 'vuetify'

import Switch, { State } from '#components/switch/Switch.vue'

const theme = useTheme()

function setTheme(value: State) {
  theme.global.name.value = value === 'left' ? 'light' : 'dark'
}
</script>
