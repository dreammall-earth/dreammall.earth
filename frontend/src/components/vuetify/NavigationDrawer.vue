<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
