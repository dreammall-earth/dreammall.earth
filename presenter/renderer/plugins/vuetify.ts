// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/lib/styles/main.sass'
import { useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
// eslint-disable-next-line import/no-namespace
import * as components from 'vuetify/lib/components/index.mjs'
// eslint-disable-next-line import/no-namespace
import * as directives from 'vuetify/lib/directives/index.mjs'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: any) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    components,
    directives,
  })
