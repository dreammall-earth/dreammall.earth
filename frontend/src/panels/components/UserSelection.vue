<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
