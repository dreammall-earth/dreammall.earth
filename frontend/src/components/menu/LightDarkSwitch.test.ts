import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'
import { Component, h } from 'vue'
import { VApp } from 'vuetify/components'

import LightDarkSwitch from './LightDarkSwitch.vue'

describe('LightDarkSwitch', () => {
  const Wrapper = () => {
    return mount(VApp, {
      slots: {
        default: h(LightDarkSwitch as Component),
      },
    })
  }

  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has light theme as default', () => {
    expect(wrapper.vm.theme.global.name.value).toBe('light')
  })

  // theme seems to be global and does not reset
  describe('click button', () => {
    beforeEach(async () => {
      await wrapper.find('button').trigger('click')
    })

    it('changes to dark theme', () => {
      expect(wrapper.vm.theme.global.name.value).toBe('dark')
    })

    describe('click buttom again', () => {
      it('changes to light theme', () => {
        expect(wrapper.vm.theme.global.name.value).toBe('light')
      })
    })
  })
})
