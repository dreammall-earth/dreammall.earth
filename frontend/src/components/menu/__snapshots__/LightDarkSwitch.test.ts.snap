// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LightDarkSwitch > renders 1`] = `
<div
  class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr"
>
  <div
    class="v-application__wrap"
  >
    
    
    <button
      class="d-flex align-center justify-center light-dark-switch"
    >
      <svg
        fill="none"
        height="15"
        viewBox="0 0 28 15"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.63636 15C5.51515 15 3.71212 14.2708 2.22727 12.8125C0.742424 11.3542 0 9.58333 0 7.5C0 5.41667 0.742424 3.64583 2.22727 2.1875C3.71212 0.729167 5.51515 0 7.63636 0H20.3636C22.4848 0 24.2879 0.729167 25.7727 2.1875C27.2576 3.64583 28 5.41667 28 7.5C28 9.58333 27.2576 11.3542 25.7727 12.8125C24.2879 14.2708 22.4848 15 20.3636 15H7.63636ZM7.63636 12.5H20.3636C21.7636 12.5 22.9621 12.0104 23.9591 11.0312C24.9561 10.0521 25.4545 8.875 25.4545 7.5C25.4545 6.125 24.9561 4.94792 23.9591 3.96875C22.9621 2.98958 21.7636 2.5 20.3636 2.5H7.63636C6.23636 2.5 5.03788 2.98958 4.04091 3.96875C3.04394 4.94792 2.54545 6.125 2.54545 7.5C2.54545 8.875 3.04394 10.0521 4.04091 11.0312C5.03788 12.0104 6.23636 12.5 7.63636 12.5ZM7.63636 11.25C8.69697 11.25 9.59848 10.8854 10.3409 10.1562C11.0833 9.42708 11.4545 8.54167 11.4545 7.5C11.4545 6.45833 11.0833 5.57292 10.3409 4.84375C9.59848 4.11458 8.69697 3.75 7.63636 3.75C6.57576 3.75 5.67424 4.11458 4.93182 4.84375C4.18939 5.57292 3.81818 6.45833 3.81818 7.5C3.81818 8.54167 4.18939 9.42708 4.93182 10.1562C5.67424 10.8854 6.57576 11.25 7.63636 11.25Z"
          fill="currentColor"
        />
      </svg>
      <svg
        fill="none"
        height="15"
        style="display: none;"
        viewBox="0 0 28 15"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.63636 15C5.51515 15 3.71212 14.2708 2.22727 12.8125C0.742424 11.3542 0 9.58333 0 7.5C0 5.41667 0.742424 3.64583 2.22727 2.1875C3.71212 0.729167 5.51515 0 7.63636 0H20.3636C22.4848 0 24.2879 0.729167 25.7727 2.1875C27.2576 3.64583 28 5.41667 28 7.5C28 9.58333 27.2576 11.3542 25.7727 12.8125C24.2879 14.2708 22.4848 15 20.3636 15H7.63636ZM7.63636 12.5H20.3636C21.7636 12.5 22.9621 12.0104 23.9591 11.0312C24.9561 10.0521 25.4545 8.875 25.4545 7.5C25.4545 6.125 24.9561 4.94792 23.9591 3.96875C22.9621 2.98958 21.7636 2.5 20.3636 2.5H7.63636C6.23636 2.5 5.03788 2.98958 4.04091 3.96875C3.04394 4.94792 2.54545 6.125 2.54545 7.5C2.54545 8.875 3.04394 10.0521 4.04091 11.0312C5.03788 12.0104 6.23636 12.5 7.63636 12.5ZM20.3636 11.25C21.4242 11.25 22.3258 10.8854 23.0682 10.1562C23.8106 9.42708 24.1818 8.54167 24.1818 7.5C24.1818 6.45833 23.8106 5.57292 23.0682 4.84375C22.3258 4.11458 21.4242 3.75 20.3636 3.75C19.303 3.75 18.4015 4.11458 17.6591 4.84375C16.9167 5.57292 16.5455 6.45833 16.5455 7.5C16.5455 8.54167 16.9167 9.42708 17.6591 10.1562C18.4015 10.8854 19.303 11.25 20.3636 11.25Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="ml-2"
      >
        $t('menu.theme.light') / $t('menu.theme.dark')
      </span>
    </button>
    
    
  </div>
</div>
`;
